[SwiftUI] WKURLSchemeHandlerAdapter leaks object identifiers and completed tasks in its tasks dictionary

pxlcoder opened the pull request:
#### 8e0ab210391c61b9ba3465e7d9d976656be23011
<pre>
[SwiftUI] WKURLSchemeHandlerAdapter leaks object identifiers and completed tasks in its tasks dictionary
<a href="https://bugs.webkit.org/show_bug.cgi?id=295531">https://bugs.webkit.org/show_bug.cgi?id=295531</a>
<a href="https://rdar.apple.com/155079754">rdar://155079754</a>

Reviewed by Patrick Angle.

* Source/WebKit/UIProcess/Cocoa/WKURLSchemeHandlerAdapter.swift:
(WKURLSchemeHandlerAdapter.webView(_:start:)):

Remove the task once finished or failed. Currently, it is only removed when stopped.

Canonical link: <a href="https://commits.webkit.org/297130@main">https://commits.webkit.org/297130@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/382584cbb71e8e39f03cc18e9785b8d5443e5956

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110554 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30213 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20646 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116580 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60821 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30892 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38801 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84072 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113502 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24664 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99547 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64513 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24026 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60375 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94047 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17746 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119370 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37594 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27911 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93037 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37968 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95818 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92860 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37867 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15607 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33567 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17853 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37489 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42960 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37152 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40491 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38860 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/UIProcess/Cocoa/WKURLSchemeHandlerAdapter.swift:
```diff
@@ -52,6 +52,8 @@ final class WKURLSchemeHandlerAdapter: NSObject, WKURLSchemeHandler {
                 urlSchemeTask.didFailWithError(error)
             }
+
+            tasks.removeValue(forKey: ObjectIdentifier(urlSchemeTask))
```
in swift, setting a value to nil is preferred over removeValue

pxlcoder committed:
[SwiftUI] WKURLSchemeHandlerAdapter leaks object identifiers and completed tasks in its tasks dictionary
https://bugs.webkit.org/show_bug.cgi?id=295531
rdar://155079754

Reviewed by Patrick Angle.

* Source/WebKit/UIProcess/Cocoa/WKURLSchemeHandlerAdapter.swift:
(WKURLSchemeHandlerAdapter.webView(_:start:)):

Remove the task once finished or failed. Currently, it is only removed when stopped.

Canonical link: https://commits.webkit.org/297130@main
